### ISSUE TITLE
EAS Build App iOS error

### ISSUE THREAD

**Issue by titanve** [expo/expo#25105]
### Summary

when building an iOS app with EAS I get the following error:
```
+ [[ ! -f /Users/expo/Library/Developer/Xcode/DerivedData/MyApp-dytmggbduaumbmbkuwfpwuxdshsg/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/BuildProductsPath/Release-iphoneos/MyApp.app/main.jsbundle ]]

+ echo 'error: File /Users/expo/Library/Developer/Xcode/DerivedData/MyApp-dytmggbduaumbmbkuwfpwuxdshsg/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/BuildProductsPath/Release-iphoneos/MyApp.app/main.jsbundle does not exist. This must be a bug with React Native, please report it here: https://github.com/facebook/react-native/issues'

error: File /Users/expo/Library/Developer/Xcode/DerivedData/MyApp-dytmggbduaumbmbkuwfpwuxdshsg/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/BuildProductsPath/Release-iphoneos/MyApp.app/main.jsbundle does not exist. This must be a bug with React Native, please report it here: https://github.com/facebook/react-native/issues

+ exit 2
```

Crossposting here for reference: https://github.com/facebook/react-native/issues/41229#issuecomment-1784205784

### What platform(s) does this occur on?

iOS

### SDK Version

47

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 14.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.18.1 - ~/.nvm/versions/node/v16.18.1/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v16.18.1/bin/yarn
      npm: 9.6.4 - ~/.nvm/versions/node/v16.18.1/bin/npm
      Watchman: 2023.02.20.00 - /opt/homebrew/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9014738
      Xcode: 15.0.1/15A507 - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 5.6.0
      expo-cli: 6.3.10
    Expo Workflow: managed

### Minimal reproducible example

eas build --platform ios

**Comment by brentvatne** [comment]
this is likely caused by a bundling error. look at the xcode build logs phase for more information. also, check out the guide for troubleshooting build errors: https://docs.expo.dev/build-reference/troubleshooting/

at the bottom of the page there is more information on how to ask a good question after you've followed the guide if you're still stuck: https://docs.expo.dev/build-reference/troubleshooting/#how-to-ask-a-good-question

**Comment by titanve** [comment]
@brentvatne there are more people having the issue, look: https://github.com/facebook/react-native/issues/41229

**Comment by brentvatne** [comment]
thanks for the report @titanve - it looks like this was a regression in a more recent patch version of react-native 0.70. i rolled back the sdk 47 recommended version to 0.70.8

**Comment by banesp** [comment]
@brentvatne this seems to fix the issue. But after downgrading back to react-native 0.70.8 I'm experiencing this issue [unary_function](https://github.com/expo/expo/issues/24608). From the react-native changelog it seems there are fixes Xcode 15 / Hermes issued in both v0.70.10 and v0.70.14.

Is there any other fix for this than downgrading to Xcode 14.3.1 or just waiting for a fix?

**Comment by brentvatne** [comment]
@banesp - if you use prebuild, you can use `npx expo prebuild --skip-dependency-update react-native` and then set the react-native version to which ever you prefer. in your build profile for eas build, that would be: `"prebuildCommand": "prebuild --skip-dependency-update react-native --non-interactive --no-install"`.

if you want to keep using JSC with SDK 47 and update to the latest react-native patch version, then you can set the env var `USE_HERMES` to `0` in your build profile and that should have the same effect of fixing the original issue here.
